Source/Diagnostics/ReducedDiags/FieldReduced.cpp

```diff
@@ -0,0 +1,257 @@
+    }
+
+    // read number of levels
+    int nLevel = 0;
+    ParmParse pp("amr");
+    pp.query("max_level", nLevel);
+    nLevel += 1;
+
+    // resize data array
+    int data_size = 0;
+    if (m_fieldEnergy) {data_size += 3*nLevel;}
+    if (m_maxField)
+    {
+        m_offset_maxField = data_size;
+        data_size += 8*nLevel;
+    }
+    m_data.resize(data_size,0.0);
+
+    if (ParallelDescriptor::IOProcessor())
+    {
+        if ( m_IsNotRestart )
+        {
+            // open file
+            std::ofstream ofs;
+            ofs.open(m_path + m_rd_name + "." + m_extension,
+                std::ofstream::out | std::ofstream::app);
+            // write header row
+            ofs << "#";
+            ofs << "[1]step()";
+            ofs << m_sep;
+            ofs << "[2]time(s)";
+
+            if (m_fieldEnergy)
+            {
+                for (int lev = 0; lev < nLevel; ++lev)
+                {
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(3+3*lev) + "]";
+                    ofs << "total_lev"+std::to_string(lev)+"(J)";
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(4+3*lev) + "]";
+                    ofs << "E_lev"+std::to_string(lev)+"(J)";
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(5+3*lev) + "]";
+                    ofs << "B_lev"+std::to_string(lev)+"(J)";
+                }
+            }
+
+            if (m_maxField)
+            {
+                for (int lev = 0; lev < nLevel; ++lev)
+                {
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(3+8*lev) + "]";
+                    ofs << "max_Ex_lev"+std::to_string(lev)+" (V/m)";
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(4+8*lev) + "]";
+                    ofs << "max_Ey_lev"+std::to_string(lev)+" (V/m)";
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(5+8*lev) + "]";
+                    ofs << "max_Ez_lev"+std::to_string(lev)+" (V/m)";
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(6+8*lev) + "]";
+                    ofs << "max_|E|_lev"+std::to_string(lev)+" (V/m)";
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(7+8*lev) + "]";
+                    ofs << "max_Bx_lev"+std::to_string(lev)+" (T)";
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(8+8*lev) + "]";
+                    ofs << "max_By_lev"+std::to_string(lev)+" (T)";
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(9+8*lev) + "]";
+                    ofs << "max_Bz_lev"+std::to_string(lev)+" (T)";
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(10+8*lev) + "]";
+                    ofs << "max_|B|_lev"+std::to_string(lev)+" (T)";
+                }
+            }
```

Comment:
If both `m_fieldEnergy` and `m_maxField` are true, would the column header numbers generated by the above if statement be wrong?
Would it be something like:
```
#[1]step() [2]time(s) [3]total_lev0(J) [4]E_lev0(J) [5]B_lev0(J) [3]max_Ex_lev0(V/m) [4]......
```
But you want `[3]max_Ex_lev0(V/m)` to be `[6]max_Ex_lev0(V/m)`.
If so, this could be fixed by using an external variable that stores how many columns have been used by the first if statement I guess.

Reply:
You're right, we should use `m_offset_maxField` here, thanks! This should be fixed now.